library/L1_Peripheral/lpc40xx/watchdog_interface.hpp

```diff
@@ -0,0 +1,19 @@
+#pragma once
+
+#include <cstdint>
+#include <iterator>
+
+namespace sjsu
+{
+
+class WatchdogInterface
+{
+ public:
+     virtual void Initialize(uint32_t timer_constant, uint32_t timer_window,
+     uint32_t timer_warning) = 0;
+     virtual void RegisterIrqHandler() = 0;

```

Comment:
I feel like this shouldn't be apart of the interface and should only be found in the implementation. Thus, this should be removed.

Reply:
Actually! What would necessary is that we need to be able to assign our own handler for the WatchDog. So when the WatchDog interrupt is triggered, our function gets called.